tests/semantic_analyser.cpp

```diff
@@ -770,7 +773,7 @@ kprobe:f { lhist(-10, -10, 10, 1); }
            ~~~~~~~~~~~~~~~~~~~~~~
 )");
   test_error("kprobe:f { @ = lhist(-10, -10, 10, 1); }", R"(
-stdin:1:16-38: ERROR: lhist() min must be non-negative (provided min -10)
```

Comment:
nit: I do think the little "(provided min -10)" is helpful

Reply:
Because the negative literals here are a separate kind, this is now folded into the case where anything at all is provided other than a non-negative integer. I could specifically re-add a case that checks to see if you've provided a negative number, but it doesn't feel worth having a special case just for that in `lhist`.

Once we converge on something generic for checking against prototypes, we could certainly include the value of the actual provided expression.